dev/MRTCore/mrt/Microsoft.Windows.ApplicationModel.Resources/src/ResourceContext.cpp

```diff
@@ -1,15 +1,76 @@
-﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Copyright (c) Microsoft Corporation and Contributors.
 // Licensed under the MIT License.
 
 #include "pch.h"
 #include "ResourceContext.h"
 #include "ResourceContext.g.cpp"
 #include "winrt/Windows.Globalization.h"
+#include <wil\resource.h>
 
 const wchar_t c_languageQualifierName[] = L"Language";
 
 namespace winrt::Microsoft::Windows::ApplicationModel::Resources::implementation
 {
+
+ResourceContext::~ResourceContext()
+{
+    winrt::slim_lock_guard guard {m_lock};
```

Comment:
Per comments below, this lock shouldn't be necessary - there should be no more references to the object when it's having its final release handled.

Reply:
removed